src/server/users/models.js

```diff
@@ -1,8 +1,155 @@
+import {KeyUtils, ENTITY, STATE} from './../key-utils';
+import bcrypt from './../bcrypt';
 import db, {fetchPrefix} from '../database';
-import {KeyUtils, ENTITY, STATE} from '../../server/key-utils';
+import request from 'request-promise';
+import config from 'config';
 
-export default class UserManager {
-  static modifyUser(key, value, cb) {
+const FACEBOOK_BASE_URL = 'https://graph.facebook.com';
+const FACEBOOK_USER_PROFILE_URL = '/me';
+const FACEBOOK_PROFILE_IMAGE_URL = '/picture';
+
+export const FacebookManager = {
+  getProfile: (accessToken) => {
+    return request({
+        uri: `${FACEBOOK_BASE_URL}${FACEBOOK_USER_PROFILE_URL}`,
+        qs: {
+          fields: 'name,email,verified',
+          access_token: accessToken
+        },
+        json: true
+      })
+  },
+  getId: (accessToken) => {
+    return FacebookManager.getProfile(accessToken)
+      .then(profile => {
+        return profile.id;
+      });
+  },
+  getProfileImage: (id) => {
+    return request({
+      uri: `${FACEBOOK_BASE_URL}/${id}${FACEBOOK_PROFILE_IMAGE_URL}`,
+      qs: { type: 'large', redirect: 0 },
+      json: true
+    });
+  },
+  getTestAccessToken: () => {
+    const APP_ID = process.env.FACEBOOK_APP_ID || config.FACEBOOK_APP_ID;
+    const APP_ACCESS_TOKEN = process.env.FACEBOOK_APP_ACCESS_TOKEN
+      || config.FACEBOOK_APP_ACCESS_TOKEN;
+    return request({
+      uri: `${FACEBOOK_BASE_URL}/v2.8/${APP_ID}/accounts/test-users`,
+      qs: { access_token: APP_ACCESS_TOKEN },
+      json: true
+    })
+    .then(res => {
+      return res.data[0].access_token;
+    });
+  }
+};
+
+export const USER_TYPE = {
+  ANONYMOUS: 0,
+  FACEBOOK: 1
+};
+
+const UserManager = {
+  addUser: (key, value) => {
```

Comment:
This function is just promise wrapping of db.put
Please use just db.put or
Add function to db directly something like 'putPromise'